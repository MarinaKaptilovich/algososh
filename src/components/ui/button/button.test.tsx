import { fireEvent, render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { Button } from './button'

it('корректный рендер кнопки с текстом', () => {
	const tree = renderer.create(<Button text='Нажать на кнопку' />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер кнопки без текста', () => {
	const tree = renderer.create(<Button />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер заблокированной кнопки', () => {
	const tree = renderer.create(<Button disabled={true} />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер кнопки с индикацией загрузки', () => {
	const tree = renderer.create(<Button isLoader={true} />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный вызов колбека при клике на кнопку', () => {
	const onClickMock = jest.fn()
	render(<Button text='Нажать на кнопку' onClick={onClickMock} />)
	fireEvent.click(screen.getByText('Нажать на кнопку'))
	expect(onClickMock).toHaveBeenCalledTimes(1)
})
