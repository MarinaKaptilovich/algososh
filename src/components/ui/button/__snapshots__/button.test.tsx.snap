// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`корректный рендер заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`корректный рендер кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`корректный рендер кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`корректный рендер кнопки с лоадером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`корректный рендер кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Нажать на кнопку
  </p>
</button>
`;

exports[`корректный рендер пустой кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
